jib-core/src/main/java/com/google/cloud/tools/jib/http/Connection.java

```diff
@@ -17,68 +17,63 @@
 package com.google.cloud.tools.jib.http;
 
 import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpHeaders;
 import com.google.api.client.http.HttpMethods;
 import com.google.api.client.http.HttpRequest;
-import com.google.api.client.http.HttpRequestFactory;
-import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpResponseException;
 import com.google.api.client.http.HttpTransport;
 import com.google.api.client.http.apache.ApacheHttpTransport;
+import com.google.cloud.tools.jib.api.LogEvent;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
-import java.io.Closeable;
 import java.io.IOException;
+import java.net.ConnectException;
 import java.net.URL;
 import java.security.GeneralSecurityException;
-import java.util.function.Function;
-import javax.annotation.Nullable;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import javax.net.ssl.SSLException;
 import org.apache.http.auth.AuthScope;
 import org.apache.http.auth.UsernamePasswordCredentials;
 import org.apache.http.impl.client.DefaultHttpClient;
 
 /**
- * Sends an HTTP {@link Request} and stores the {@link Response}. Clients should not send more than
- * one request.
- *
- * <p>Example usage:
- *
- * <pre>{@code
- * try (Connection connection = new Connection(url)) {
- *   Response response = connection.get(request);
- *   // ... process the response
- * }
- * }</pre>
+ * Thread-safe HTTP client with the automatic insecure connection failover feature. Intended to be
+ * created once and shared to be called at multiple places. Callers should close the returned {@link
+ * Response}.
  */
-public class Connection implements Closeable {
+public class Connection { // TODO: rename to TlsFailoverHttpClient
 
-  /**
-   * Returns a factory for {@link Connection}.
-   *
-   * @return {@link Connection} factory, a function that generates a {@link Connection} to a URL
-   */
-  public static Function<URL, Connection> getConnectionFactory() {
+  private static boolean isHttpsProtocol(URL url) {
+    return "https".equals(url.getProtocol());
+  }
+
+  private static URL toHttp(URL url) {
+    GenericUrl httpUrl = new GenericUrl(url);
+    httpUrl.setScheme("http");
+    return httpUrl.toURL();
+  }
+
+  private static HttpTransport getHttpTransport() {
```

Comment:
Given we have `getInsecureHttpTransport()`, would this make sense as `getSecureHttpTransport()`?